### PR TITLE
feat: zoom and pipelines

### LINKED ISSUE #1087: zoom is not centered

### PR DESCRIPTION
**Pull request by hugomrdias**
fix: improve pipelines watcher

feat: lightbox for images

feat: add watch

feat: fix links, support diagrams in images and pipelines

   Pipelines support mmd and dot Links now show .Page.Title in the link text
   if available Remove figure, h, svg shortcodes, we can just use  now

fix: switch to d3 zoomable

closes #1087

### CODE REVIEW

File: tools/diagrams.js [hunk 1]
```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env node
+
+const globby = require('globby');
+const execa = require('execa')
+const path = require('path')
+const fs = require('fs')
+const chokidar = require('chokidar');
+
+const runMmd = (p) => {
+    const outDir =path.dirname(p).replace('content/', 'static/_gen/diagrams/')
```
Comment (olizilla) [inline tools/diagrams.js:10]: olint: the olizilla linter
```suggestion
    const outDir = path.dirname(p).replace('content/', 'static/_gen/diagrams/')
```

File: tools/diagrams.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    ], { preferLocal: true });
+}
+
+const runMmdAll = async () => {
+	const paths = await globby(['content/**/*.mmd']);
+    await Promise.all(paths.map(runMmd))
+}
+
+const runDot = (p) => {
+    const outDir =path.dirname(p).replace('content/', 'static/_gen/diagrams/')
```
Comment (olizilla) [inline tools/diagrams.js:27]:
```suggestion
    const outDir = path.dirname(p).replace('content/', 'static/_gen/diagrams/')
```

File: README.md [hunk 3]
```diff
@@ -179,15 +131,30 @@ Random plain text context ...
...
+For diagrams you can use the source files and the pipelines will handle converting that to `svg`, we support mermaid and dot source files.
+
+```md
+# relative to the markdown file
+![Alt text](picture.jpg)
+
+# relative to the content folder
+![Alt text](/content/intro/diagram1.mmd)
+
+![Alt text](graph.dot "Graph title")
```
Comment (olizilla) [inline README.md:145]:
```suggestion
# relative to the markdown file
![Alt text](picture.jpg)
![Alt text](graph.dot "Graph title")

# relative to the content folder
![Alt text](/content/intro/diagram1.mmd)
```

**Review by olizilla — DISMISSED** [review]
This is rad. I notice that I don't get live-reload on image changes... i tried editing a dot file and it correctly rebuilt but livereload did not kick in. Changes to html do trigger it. Do you see that too?

**Comment by hugomrdias** [comment]
> This is rad. I notice that I don't get live-reload on image changes... i tried editing a dot file and it correctly rebuilt but livereload did not kick in. Changes to html do trigger it. Do you see that too?

you probably still have the SW registered, remove it and it will not register again in dev

**Comment by olizilla** [comment]
Something is wonky with the image loading on the prod site

Screenshot 2020-08-26 at 14 53 04

File: package.json [hunk 4]
```diff
@@ -4,10 +4,12 @@
   "description": "",
   "main": "index.js",
   "scripts": {
+    "start": "concurrently -n diagrams,hugo \"tools/diagrams.js --watch\" \"npm run serve\"",
     "test": "echo \"Error: no test specified\" && exit 1",
-    "serve": "hugo server --bind=0.0.0.0 --disableFastRender",
+    "serve": "tools/diagrams.js --all && hugo server --bind=0.0.0.0 --disableFastRender",
     "build": "hugo --gc --minify",
```
Comment (olizilla) [inline package.json:10]:
```suggestion
    "build": "tools/diagrams.js && hugo --gc --minify",
```

Comment (olizilla) [reply package.json:10]: need to build the diagrams as part of build so all works in CI

**Comment by olizilla** [comment]
@hugomrdias you like this? https://github.com/filecoin-project/specs/pull/1104

**Pull request merged**